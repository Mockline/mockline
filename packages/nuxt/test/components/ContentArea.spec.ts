import { describe, it, expect } from 'vitest'
// @ts-ignore
import {MContentArea} from "#components";
import ComponentRender from "../component-render";

describe('ContentArea', () => {
  it.each([
    ['base case', {}],
    ['with margin', { props: { margin: false } }],
    ['without margin', { props: { margin: true } }],
    ['<MContentArea margin>Super Slot</MContentArea>>', { props: { margin: true } }],
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MContentArea.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MContentArea)
    expect(html).toMatchSnapshot()
  })
})
