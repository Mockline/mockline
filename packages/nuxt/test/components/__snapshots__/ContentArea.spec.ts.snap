// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentArea > renders <MContentArea margin>Super Slot</MContentArea>> correctly 1`] = `"<div class="overflow-y-auto relative rounded-[calc(var(--radius-base)*2)] bg-content-area border border-content-area-border h-[calc(100vh-var(--spacing)*2)] m-1"></div>"`;

exports[`ContentArea > renders base case correctly 1`] = `"<div class="overflow-y-auto relative rounded-[calc(var(--radius-base)*2)] bg-content-area border border-content-area-border h-[calc(100vh-var(--spacing)*2)] m-1"></div>"`;

exports[`ContentArea > renders with margin correctly 1`] = `"<div class="h-[100vh] overflow-y-auto relative rounded-[calc(var(--radius-base)*2)] bg-content-area border border-content-area-border"></div>"`;

exports[`ContentArea > renders without margin correctly 1`] = `"<div class="overflow-y-auto relative rounded-[calc(var(--radius-base)*2)] bg-content-area border border-content-area-border h-[calc(100vh-var(--spacing)*2)] m-1"></div>"`;
